### PR TITLE
Add regex compare op

### PR DESCRIPTION
**Pull request by wwwil**
I've added a `regex` compare op to allow a test to check if a flag or path value matches a given regular expression. This makes use of the Go "regexp" package which was already being used.

I've added this a I'm planning to used kube-bench to run some of my own custom test suites which needed regex. It may also be helpful for running other tests, and should not affect any other existing functionality.

### CODE REVIEW

**Comment by ttousai** [comment]
@wwwil that is a wonderful idea. The code looks good to me.

But what do you think about calling the operator `match`, it seems more intuitive than regex. Anyone reading the test_item will see you want to match against a regexp.

@lizrice wdyt?

**Comment by lizrice** [comment]
Thanks for the idea @ttousai but I’m happy with the name “regex”. I think people might assume that “match” is the same as “eq”.

File: README.md [hunk 1]
```diff
@@ -228,6 +228,9 @@ These operations are:
 - `lte`: tests if the flag value is less than or equal to the compared value.
 - `has`: tests if the flag value contains the compared value.
 - `nothave`: tests if the flag value does not contain the compared value.
+- `regex`: tests if the flag value matches the compared value regular expression.
+
+When defining regular expressions in YAML it is generally easier to wrap them in single quotes, for example `'^[abc]$`, to avoid issues with string escaping.
```
[the remark this answers is not in the document]
Comment (lizrice) [reply README.md]: I’m going to go ahead and edit this assuming I’m right, so we can get this merged

[the remark this answers is not in the document]
Comment (wwwil) [reply README.md]: Yeah that was missing a single quote! Thanks for fixing and merging 😄

**Pull request merged**